assets/js/blocks/checkout/input/index.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { registerBlockType } from '@wordpress/blocks';
+import { TextControl } from '@wordpress/components';
+
+registerBlockType( 'woocommerce/checkout-input', {
+	title: __( 'Checkout input', 'woo-gutenberg-products-block' ),
+	category: 'woocommerce-checkout',
+	keywords: [ __( 'WooCommerce', 'woo-gutenberg-products-block' ) ],
+	supports: {
+		html: false,
+	},
+	attributes: {
+		label: {
+			type: 'string',
+			default: '',
+		},
+		type: {
+			type: 'string',
+			default: '',
+		},
+	},
+	edit( { attributes } ) {
+		const { label, type } = attributes;
+
+		return (
+			<TextControl
+				disabled
+				label={ label }
+				type={ type }
```

Comment:
Will we ever change `type` here?

Reply:
Besides `text`, the default form uses `tel`, `email` and `password` types. AFAIK all render the same in all main browsers, but I still thought it made sense honoring the type considering some times browsers add some differences depending on the type (thinking on spinners of `type="button"` inputs). Does it make sense?